### PR TITLE
Basic game functionality [WIP]

### PR DESCRIPTION
**Pull request by TheFehr**
- [ ] incoming message validation

- [ ] game listing

- [ ] game join

- [ ] game setStone

### CODE REVIEW

File: backend/client.go [hunk 1]
```diff
@@ -250,13 +283,18 @@ func serverNewGameHandlder(c *client, recMessage message) {
 	}
 }
 
+func gameJoinHandler(c *client, recMessage message) {
+
```
Comment (lukasbischof) [inline backend/client.go]: This is empty

Comment (skius) [reply backend/client.go]: I agree

**Review by skius — CHANGES_REQUESTED** [review]

File: backend/client.go [hunk 2]
```diff
@@ -69,44 +69,69 @@ func parseMessage(raw string) (m message, err error) {
...
+type messageValidationError struct {
+	err            string
+	returnToClient bool
+}
+
+func (messageErr *messageValidationError) Error() string {
+	return messageErr.err
+}
+
+func validateMessage(received message, expected message) (errs []error) {
```
Comment (skius) [inline backend/client.go]: Great idea, return value as []error

**Review by lukasbischof — CHANGES_REQUESTED** [review]
Please implement @skius changes

**Review by skius — APPROVED** [review]
About the _slice_

File: backend/server/player.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	stoppedChan chan bool
+	receiveChan chan playerMessage
+	sendChan    chan serverMessage
+	state       ClientState
+	server      *Server
+	currentGame uuid.UUID
+}
+
+func newPlayer(name string, conn net.Conn, server *Server) (pl *player) {
+	user = &Types.User{ugins
```
Comment (lukasbischof) [inline backend/server/player.go:27]: What's an `ugin`?

**Review by lukasbischof — CHANGES_REQUESTED** [review]

**Pull request closed without merging**